agent-tooling/src/main/java/io/opentelemetry/auto/tooling/muzzle/ReferenceMatcher.java

```diff
@@ -293,4 +300,19 @@ private static boolean matchesPrimitive(String longName, String shortName) {
     }
     return null;
   }
+
+  private static List<Mismatch> lazyAdd(List<Mismatch> mismatches, Mismatch mismatch) {
```

Comment:
What is the motivation of this optimisation?

Reply:
I suggest you review the original PR.

Reply:
added comment